pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -775,16 +837,24 @@ func (nc *Controller) monitorNodeHealth() error {
 			zoneToNodeConditions[utilnode.GetZoneKey(node)] = append(zoneToNodeConditions[utilnode.GetZoneKey(node)], currentReadyCondition)
 		}
 
-		nodeHealthData := nc.nodeHealthMap.getDeepCopy(node.Name)
-		if nodeHealthData == nil {
-			klog.Errorf("Skipping %v node processing: health data doesn't exist.", node.Name)
-			continue
-		}
 		if currentReadyCondition != nil {
+			pods, err := nc.getPodsAssignedToNode(node.Name)
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("unable to list pods of node %v: %v", node.Name, err))
+				if currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue {
```

Comment:
good catch
that said, it requires some unit test

Reply:
New case to `TestMonitorNodeHealthMarkPodsNotReadyRetry` added.